src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/stores/DatagridStore.js

```diff
@@ -9,30 +9,51 @@ export default class DatagridStore {
     @observable data: Array<Object> = [];
     @observable selections: Array<string | number> = [];
     @observable loading: boolean = true;
+    @observable reset: boolean = false;
+    @observable appendRequestData: boolean = false;
     disposer: () => void;
     resourceKey: string;
     options: Object;
     observableOptions: ObservableOptions;
-    appendRequestData: boolean;
     localeInterceptDisposer: () => void;
-    dataObserveDisposer: () => void;
+
+    static getAppendRequestData: (loadingStrategy: string) => boolean = (loadingStrategy: string) => {
+        switch (loadingStrategy) {
+            case 'infiniteScroll':
+                return true;
+            default:
+                return false;
+        }
+    };
 
     constructor(
         resourceKey: string,
         observableOptions: ObservableOptions,
-        options: Object = {},
-        appendRequestData: boolean = false
+        options: Object = {}
     ) {
         this.resourceKey = resourceKey;
         this.observableOptions = observableOptions;
         this.options = options;
+    }
+
+    @action init = (loadingStrategy: string) => {
```

Comment:
Instead of this additional constructor we could still make the `autorun` in the constructor, and build something like a guard in the `sendRequest` method, couldn't we? Would then look something like this:

```javascript
sendRequest = () => {
    if (!this.loadingStrategy) {
        return;
    }
}
```

The only thing we would have to take care of, is that this single `if` includes all strategies that are mandatory in order to do a request.